apps/frontpage/components/docs/renderers/index.tsx

```diff
@@ -0,0 +1,70 @@
+'use client';
+
+import { renderers } from '@utils';
+import { FC } from 'react';
+import { useDocs } from '../../../app/docs/provider';
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from '@ui';
+import { Button } from './button';
+
+interface RenderersProps {
+  activeRenderer: string;
+}
+
+export const Renderers: FC<RenderersProps> = () => {
+  const { activeRenderer, setRenderer } = useDocs();
+
+  const firstThreeRenderers = renderers.slice(0, 3);
+  const restRenderers = renderers.slice(4);
+  const isInFirstThree = firstThreeRenderers.some(
+    (renderer) => renderer.id === activeRenderer
+  );
+  const activeRendererObj = renderers.find(
+    (renderer) => renderer.id === activeRenderer
+  );
+
+  return (
+    <div className="flex gap-2 mb-8">
+      {firstThreeRenderers.map((renderer) => (
+        <Button
+          key={renderer.id}
+          active={renderer.id === activeRenderer}
+          onClick={() => setRenderer(renderer.id)}
+          title={renderer.title}
+        />
+      ))}
+      {!isInFirstThree && activeRendererObj ? (
+        <Button
+          active={activeRendererObj.id === activeRenderer}
+          onClick={() => setRenderer(activeRendererObj.id)}
+          title={activeRendererObj.title}
+        />
+      ) : (
+        <Button
+          active={renderers[3].id === activeRenderer}
+          onClick={() => setRenderer(renderers[3].id)}
+          title={renderers[3].title}
+        />
+      )}
+      <DropdownMenu>
+        <DropdownMenuTrigger asChild>
+          <Button title="More" arrow />
+        </DropdownMenuTrigger>
+        <DropdownMenuContent>
+          {restRenderers.map((renderer) => (
```

Comment:
`restRenderers` will need to be updated to include the 4th renderer (i.e. `Web components`) when the active renderer is not one of the first four.

For example, in this scenario, where `Svelte` is active:

![](https://github.com/storybookjs/web/assets/486540/1e0d414e-dba1-478c-b751-362a60ad6c7c)

The first renderer in the menu should be `Web components`.

You can find all of the supported states here:

- https://main--5be26744d2f6250024a9117d.chromatic.com/?path=/story/screens-docsscreen-rendererselector--community-renderer-selected
- https://github.com/storybookjs/frontpage/blob/main/src/components/screens/DocsScreen/RendererSelector.tsx

Reply:
Ah good call. I forgot about that. I'm adding a fix to it now.

Reply:
This is now fixed.